src/cohort.js

```
@@ -0,0 +1,104 @@
+class Cohort {
+  constructor() {
+    this.cohortList = []
+  }
+
+  createCohort(cohortName) {
+    const newCohort = { name: cohortName, students: [], capacity: 24 }
+    const findCohort = this.cohortList.find(
+      (cohort) => cohort.name === cohortName
+    )
+
+    if (findCohort) {
+      throw new Error('This Cohort already exists, please choose another name!')
+    }
+    if (!cohortName) {
+      throw new Error('Please give a name to your class!')
+    } else {
+      this.cohortList.push(newCohort)
+    }
+
+    return this.cohortList
+  }
+
+  searchCohort(cohortName) {
+    const findCohort = this.cohortList.find(
+      (cohort) => cohort.name === cohortName
+    )
+
+    if (!findCohort) {
+      throw new Error('Cohort not found')
+    }
+
+    return findCohort
+  }
+
+  searchStudent(cohortName, studentID) {
+    const findCohort = this.cohortList.find(
+      (cohort) => cohort.name === cohortName
+    )
+    const findStudent = findCohort.students.find(
+      (student) => student.studentID === studentID
+    )
+
+    if (!findStudent) throw new Error('Student not found')
+
+    return findStudent
+  }
+
+  addStudent(cohortName, studentToAdd) {
+    const findCohort = this.cohortList.find(
+      (cohort) => cohort.name === cohortName
+    )
+    const studentExists = findCohort.students.find(
+      (student) => student.studentID === studentToAdd.studentID
+    )
+
+    if (studentExists) {
+      throw new Error('this student already exists, add another student!')
+    }
+
+    for (let i = 0; i < this.cohortList.length; i++) {
+      for (let j = 0; j < this.cohortList[i].students.length; j++) {
+        if (studentToAdd.userName === this.cohortList[i].students[j].userName) {
+          return 'this student already enrolled in another Cohort!'
```

I think this would make more sense throwing an error rather than returning. All your other bad cases are throwing errors.